sdk/cdsclient/client_queue.go

```diff
@@ -50,101 +52,86 @@ func shrinkQueue(queue *sdk.WorkflowQueue, nbJobsToKeep int) time.Time {
 }
 
 func (c *client) QueuePolling(ctx context.Context, jobs chan<- sdk.WorkflowNodeJobRun, pbjobs chan<- sdk.PipelineBuildJob, errs chan<- error, delay time.Duration, graceTime int, modelType string, ratioService *int, exceptWfJobID *int64) error {
-	t0 := time.Unix(0, 0)
 	jobsTicker := time.NewTicker(delay)
-	pbjobsTicker := time.NewTicker(delay * 10)
-	oldJobsTicker := time.NewTicker(delay * 5)
+	pbjobsTicker := time.NewTicker(delay * 2)
+
+	// This goroutine call the SSE route
+	chanSSEvt := make(chan SSEvent)
+	sdk.GoRoutine("RequestSSEGet", func() {
+		for ctx.Err() == nil {
+			if err := c.RequestSSEGet(ctx, "/events", chanSSEvt); err != nil {
+				log.Println(err)
+			}
+			time.Sleep(5 * time.Second)
+		}
+	})
 
 	for {
 		select {
 		case <-ctx.Done():
-			jobsTicker.Stop()
 			pbjobsTicker.Stop()
-			oldJobsTicker.Stop()
+			jobsTicker.Stop()
 			if jobs != nil {
 				close(jobs)
 			}
 			if pbjobs != nil {
 				close(pbjobs)
 			}
 			return ctx.Err()
-		case <-oldJobsTicker.C:
-			if c.config.Verbose {
-				fmt.Println("oldJobsTicker")
-			}
-
-			if jobs != nil {
-				queue := sdk.WorkflowQueue{}
-				if _, err := c.GetJSON("/queue/workflows", &queue); err != nil {
-					errs <- sdk.WrapError(err, "Unable to load old jobs")
-				}
+		case evt := <-chanSSEvt:
+			content, _ := ioutil.ReadAll(evt.Data)
+
+			var apiEvent sdk.Event
+			_ = json.Unmarshal(content, &apiEvent) // ignore errors
+			// filter only EventRunWorkflowJob
+			if apiEvent.EventType == "sdk.EventRunWorkflowJob" {
+				jobRunID, ok := apiEvent.Payload["ID"].(float64)
+				status, okStatus := apiEvent.Payload["Status"].(string)
+				if ok && okStatus && status == sdk.StatusWaiting.String() {
+					// wait for the grace time before pushing the job in the channel
+					go func() {
+						time.Sleep(time.Duration(graceTime) * time.Second)
+						job, err := c.QueueJobInfo(int64(jobRunID))
```

Comment:
Job received by SSE should contains modelType and ContainsService attributes. 
So, an hatchery openstack avoid call QueueJobInfo for each job type docker,
a hatchery marathon avoid cal QueueJobInfo for each job containsService, etc...

This will be the same behaviour with existing call /queue/workflow, with queryParams modelType and ratioService

Reply:
#3390